Sage/carbon
# fix: remove border style from input when disabled

fix #4769

### Proposed behaviour

This PR make sure the orange border style from the `Input`, and all components based on it, is rendered when they are focused. 

### Current behaviour

Currently, the input removes its border when it gets disabled but, after enabling it programmatically, the border appears without the `Input` being focused.
![image](https://user-images.githubusercontent.com/11317965/151970743-05768548-3e76-4cca-9030-9869be5530e2.png)


### Checklist




#### QA


### Additional context



### Testing instructions

From now on, we will refer to any component that contains a `Input` as `Input`. In a document with several carbon elements, follow the next steps to reproduce:

1. Focus the `Input`
2. Disable the `Input`
3. Press tab, to move the focus to the next element
3. Enable the `Input` programmatically

You'll see both elements (the `Input` and the element focused) have an orange border. See also screenshot above.

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/hidden-night-e406i?file=/src/index.js
